Webapp/sources/blueprints/main/routes.py

```diff
@@ -33,38 +34,34 @@ def index() -> Response:
     messages_from_redirects = (
         [request.args.get("message")] if request.args.get("message") else []
     )
-    # gets jinja variables if user is logged in to populate the homepage, else the non-logged in user homepage is loaded
+
+    # get default jinja variables for landing page
+    user_confirmed = None
+    form = LoginForm()
+    user_role = None
+    workgroups = []
+    notification_number = 0
+    news_items = []
+
+    if request.method == "POST":
+        # return redirects from login verification to prevent form resubmission
+        page_redirect = services.auth.verify_login(form)
+        return page_redirect
+
+    # update jinja variables if user is logged in to populate the homepage
     if current_user.is_authenticated:
-        # get the users role to determine if they can access the admin dashboard button or not
-        user = (
-            db.session.query(models.User)
-            .filter(models.User.email == current_user.email)
-            .first()
-        )
-        user_confirmed = user.is_verified
-        user_role = user.Role.name
+        form = None
+        user_role = current_user.Role.name
+        user_confirmed = current_user.is_verified
         workgroups = get_workgroups()
```

Comment:
I don't think we need the workgroups/notification number here if its just for the navbar. The inject_session_context in sources/_init__.py should do these for all pages now.

Reply:
Removed workgroups and notification number from here now. Notification number was missing from inject_session_context so I've added it